packages/xlsx-import/src/mappers/jsonMapper.ts

```diff
@@ -1,9 +1,33 @@
-type JsonMapper = <TJsonResult>(value: string) => TJsonResult | null;
 
-export const jsonMapper: JsonMapper = <TJsonResult extends any>(value: string): TJsonResult | null => {
-    try {
-        return JSON.parse(value);
-    } catch (e) {
-        return null;
+type callSignature = <TJsonResult, TDefaultResult>(value: string) => TJsonResult | TDefaultResult;
+
+type JsonMapper = {
+    default: <TDefaultResult>(defaultResult: TDefaultResult) => JsonMapper
+} & callSignature;
+
+interface IJsonMapperOptions {
+    default: any;
+}
+
+const factory = (options: Readonly<IJsonMapperOptions>): JsonMapper => {
+    const mapper = (json: string) => {
```

Comment:
I would like to propose a little update for typing for you solution:
```suggestion
type JsonMapperSignature<TDefaultResult> = <TJsonResult>(value: string) => TJsonResult | TDefaultResult;

export type JsonMapper<TJsonResult> = JsonMapperSignature<TJsonResult> & {
    default: <TDefaultResult>(value: TDefaultResult) => JsonMapper<TDefaultResult>;
};

interface IJsonMapperOptions<TDefaultResult> {
    default: TDefaultResult;
}

const factory = <TDefaultResult>(options: Readonly<IJsonMapperOptions<TDefaultResult>>): JsonMapper<TDefaultResult> => {
    const mapper: JsonMapper<TDefaultResult> = <TJsonResult>(json: string): TJsonResult | TDefaultResult => {
```

Together with the next two proposals, will enable using typing features of the mapper (more info in review comment)

Reply:
Thanks for the feedback! Will update it